cli/functionary/config.py

```diff
@@ -0,0 +1,59 @@
+from pathlib import Path
+
+import click
+from dotenv import get_key, set_key
+
+
+def save_config_value(key, value):
+    """
+    Save configuration key and value to config file. If key is already present,
+    overwrite the current value.
+
+    Args:
+        key and value to be stored as strings
+
+    Returns:
+        Nothing
+
+    Raises:
+        ClickException from PermissionError if cannot read config file
+    """
+    try:
+        functionary_dir = Path.home() / ".functionary"
+        if not functionary_dir.exists():
+            functionary_dir.mkdir()
+
+        config_file = functionary_dir / "config"
+        set_key(
+            config_file,
+            key,
+            value,
+        )
+    except PermissionError:
+        raise click.ClickException("Config file present, but could not be read")
```

Comment:
In this case, the error is writing the file rather than reading.  But in either case, I think both this and the other instance of this message could be:

```
raise click.ClickException(f"Failed to open {config_file}: Permission Denied")
```

That more clearly communicates that the user should go check the permissions on a specific file.